helix-term/src/ui/editor.rs

```diff
@@ -980,6 +980,23 @@ impl EditorView {
                     return EventResult::Consumed(None);
                 }
 
+                let result = editor.tree.views().find_map(|(view, _focus)| {
+                    view.gutter_coords_at_screen_coords(row, column)
+                        .map(|coords| (coords.0, coords.1, view.id))
+                });
+
+                if let Some((line, _, view_id)) = result {
+                    editor.tree.focus = view_id;
+
+                    let doc = &mut editor.documents[editor.tree.get(view_id).doc];
+                    if let Ok(pos) = doc.text().try_line_to_char(line) {
+                        doc.set_selection(view_id, Selection::point(pos));
+                        commands::dap_toggle_breakpoint(cxt);
```

Comment:
This should be factored out to a new function that takes an editor (not cx) and also takes a pos/range as an input. That way we're not changing the selection when setting breakpoints.

Reply:
Yes, will do!